adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -490,6 +523,19 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
             Logger.e(TAG, e.getMessage(), "", ADALError.ENCODING_IS_NOT_SUPPORTED, e);
             throw e;
         } catch (IOException e) {
+            //retry once if there is an observation of a network timeout by the client 
+            if (e instanceof SocketTimeoutException){
```

Comment:
>if (e instanceof SocketTimeoutException) [](start = 12, length = 40)

Please don't do that. Can you not catch SocketTimeoutException before this and do it in there?